src/store/single.rs

```diff
@@ -0,0 +1,116 @@
+// Copyright 2018 Mozilla
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may not use
+// this file except in compliance with the License. You may obtain a copy of the
+// License at http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software distributed
+// under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+// CONDITIONS OF ANY KIND, either express or implied. See the License for the
+// specific language governing permissions and limitations under the License.
+
+use lmdb;
+
+use lmdb::{
+    Cursor,
+    Database,
+    Iter as LmdbIter,
+    RoCursor,
+    RwTransaction,
+    Transaction,
+};
+
+use lmdb::WriteFlags;
+
+use crate::error::StoreError;
+
+use crate::value::Value;
+
+fn read_transform(val: Result<&[u8], lmdb::Error>) -> Result<Option<Value>, StoreError> {
```

Comment:
Nit: the two implementations of _read_transform()_ in single.rs and multi.rs are identical and could be refactored into their super module store.rs.